controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java

```diff
@@ -320,6 +320,45 @@ public void deleteStream(final String scopeName, final String streamName, final
         .thenAccept(x -> LoggerHelpers.traceLeave(log, "deleteStream", traceId));
     }
 
+    @Override
+    public void deleteReaderGroup(final String scopeName, final String readerGroupName,
+                                  final SecurityContext securityContext, final AsyncResponse asyncResponse) {
+        long traceId = LoggerHelpers.traceEnter(log, "deleteReaderGroup");
+        long requestId =  requestIdGenerator.nextLong();
+
+        try {
+            restAuthHelper.authenticateAuthorize(
+                    getAuthorizationHeader(),
+                    authorizationResource.ofReaderGroupInScope(scopeName, readerGroupName),
+                    READ_UPDATE);
+        } catch (AuthException e) {
+            log.warn(requestId, "delete reader group for {} failed due to authentication failure.",
+                    scopeName + "/" + readerGroupName);
+            asyncResponse.resume(Response.status(Status.fromStatusCode(e.getResponseCode())).build());
+            LoggerHelpers.traceLeave(log, "deleteReaderGroup", traceId);
+            return;
+        }
+        ClientFactoryImpl clientFactory = new ClientFactoryImpl(scopeName, this.localController, this.clientConfig);
```

Comment:
Question: creating a `ClientFactoryImpl` and a `ReaderGroupManager` on a per-call basis is expensive (aslo, you don't seem to be closing the resources). I see that in the previous call, deleting a Stream is done by `controllerService.deleteStream()` call. Why don't you do just the same calling to `deleteReaderGroup` in `ControllerService`?